src/main/resources/com/blackwaterpragmatic/webservices/mybatis/mapper/UserMapper.xml

```diff
@@ -2,15 +2,29 @@
 <!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
 <mapper namespace="com.blackwaterpragmatic.webservices.mybatis.mapper.UserMapper">
 
-	<select id="list" resultType="User">
-		select id, name
-		from Users
-	</select>
+    <select id="list" resultType="User">
```

Comment:
I've no doubt this isn't performant, but the scenario I was thinking was with an Auto-complete drop down where the user starts typing and then we async some matches based on the given string?  The requirements did not specify EXACT matches ;-)